src/components/TimeQuerier/index.tsx

```diff
@@ -0,0 +1,142 @@
+  } = TimeQuerierStore.useContainer();
+  const { defaultStepValue, minStepValue, dateFormat, timeFormat } = constants;
+  const [startTime, setStartTime] = useState(
+    moment(timeQuerierState.selectedStartTimestamp)
+  );
+  const [endTime, setEndTime] = useState(
+    moment(timeQuerierState.selectedEndTimestamp)
+  );
+  const [stepTime, setStepTIme] = useState(defaultStepValue);
+  const [error, setError] = useState<undefined | string>(undefined);
+
+  const handleStartChange = (date: moment.Moment | string) => {
+    console.log(date);
+    if (typeof date !== "string") {
+      setStartTime(date);
+      setError(undefined);
+    }
+  };
+  const handleEndChange = (date: moment.Moment | string) => {
+    if (typeof date !== "string") {
+      setEndTime(date);
+      setError(undefined);
+    }
+  };
+  const handleStepChange = (valueAsString: string, valueAsNumber: number) => {
+    if (valueAsString !== "" && valueAsNumber >= minStepValue)
+      setStepTIme(valueAsNumber);
+  };
+  const valid = (current: moment.Moment) => {
+    return current.isBefore(moment());
+  };
+  const handleFetch = () => {
+    if (
+      endTime.isBefore(startTime) ||
+      endTime.isAfter(moment()) ||
+      stepTime < minStepValue
+    )
+      setError("Kindy check you input");
+    else {
+      changeTimeQuerier(
+        startTime.toISOString(),
+        endTime.toISOString(),
+        stepTime
+      );
+    }
+  };
+
+  return (
+    <VStack mt={4} w="100%">
+      <HStack pb="1" w="100%" justifyContent="space-between">
+        <Text>Start Time</Text>
+
+        <Box sx={styles}>
+          <Datetime
+            value={startTime}
+            dateFormat={dateFormat}
+            timeFormat={timeFormat}
+            inputProps={{ className: "custom-datepicker" }}
+            onChange={handleStartChange}
+            isValidDate={valid}
+          />
+        </Box>
+      </HStack>
+      <HStack pb="1" w="100%" justifyContent="space-between">
+        <Text>End Time</Text>
+        <Box sx={styles}>
+          <Datetime
+            value={endTime}
+            dateFormat={dateFormat}
+            timeFormat={timeFormat}
+            inputProps={{ className: "custom-datepicker" }}
+            onChange={handleEndChange}
+            isValidDate={valid}
+            data-testid="end-time"
+          />
+        </Box>
+      </HStack>
+      <HStack pb="1" w="100%" justifyContent="space-between">
+        <Text>Step Time (s)</Text>
+        <NumberInput
+          w="60%"
+          size="sm"
+          value={stepTime}
+          onChange={handleStepChange}
+          min={minStepValue}
+          data-testid="step-value"
+        >
+          <NumberInputField />
+          <NumberInputStepper>
+            <NumberIncrementStepper />
+            <NumberDecrementStepper />
+          </NumberInputStepper>
+        </NumberInput>
+      </HStack>
+      <HStack pt="2" pb="1" w="100%">
+        <Button
+          onClick={handleFetch}
```

Comment:
rename `handleFetch` to some appropriate name

Reply:
Actually this fetches the data for a particular start end date I thought this name conveys that message any suggestions?

Reply:
hanldeFetchTimeSeriesData